docs/faq.md

```diff
@@ -0,0 +1,50 @@
+### Regions = Placeholders?
+
+Yes. Regions can be understood as placeholders, i.e. sidebars, footer etc. It create a node to let the child widgets to mount onto.
+
+Core App would have regions created on them; it does not know what will be rendered on that spot:
+```
+<div>
+ <Region name="sidebar" />
+</div>
+```
+
+Child widgets would then register themselves onto a region. It tells the child widget where it should appear on the page:
+
+```
+childWidget.setRegion('sidebar');
+```
+
+Please note that normally developers would not need to worry about creating regions on the core app; this is handled behind the scenes. You will only need to know how to mount child widgets on to pre-specified regions.
```

Comment:
this part is related to Fireball. and is wiser to not include it here. we would have a separate set of documentation for that soon.